### PR TITLE
docs: LDAP user provider

### PR DESCRIPTION
**Pull request by MichaelScofield**
## What's Changed in this PR

Docs about LDAP user provider.

## Checklist

- [x] Please confirm that all corresponding versions of the documents have been revised.
- [x] Please ensure that the content in `sidebars.ts` matches the current document structure when you changed the document structure.
- [ ] This change requires follow-up update in localized docs.

### CODE REVIEW

File: docs/user-guide/deployments/authentication/ldap.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+:::tip NOTE
+
+**LDAP user provider is only available in GreptimeDB Enterprise.**
+
+:::
+
+# LDAP user provider configuration
+
+GreptimeDB can connect to an external LDAP server to authenticate users. We borrow similar ideas
+from [PostgreSQL](https://www.postgresql.org/docs/current/auth-ldap.html). In GreptimeDB, LDAP authentication is
```
Comment (sunng87) [inline docs/user-guide/deployments/authentication/ldap.md:12]:
```suggestion
GreptimeDB can connect to an external LDAP server to authenticate users. Similar to [LDAP in PostgreSQL](https://www.postgresql.org/docs/current/auth-ldap.html), in GreptimeDB, LDAP authentication is
```

File: docs/user-guide/deployments/authentication/overview.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+- [Static user provider](./static.md): a simple built-in user provider implementation that finds users from a static
+  file.
+- [LDAP user provider](./ldap.md): a user provider implementation that authenticates users against an external LDAP
+  server.
+
+:::tip NOTE
+
+**LDAP user provider is only available in GreptimeDB Enterprise.**
+
+:::
```
Comment (sunng87) [inline docs/user-guide/deployments/authentication/overview.md:15]:
```suggestion
- [LDAP user provider](./ldap.md): **Enterprise feature.** A user provider implementation that authenticates users against an external LDAP
  server.

```

File: docs/user-guide/deployments/authentication/overview.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# Authentication
+
+Authentication occurs when a user attempts to connect to the database. In GreptimeDB, users are authenticated by "user
+provider"s. There are various implementations of user providers in GreptimeDB:
+
+- [Static user provider](./static.md): a simple built-in user provider implementation that finds users from a static
```
Comment (sunng87) [inline docs/user-guide/deployments/authentication/overview.md:6]:
```suggestion
- [Static user provider](./static.md): A simple built-in user provider implementation that finds users from a static
```

File: i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/authentication/ldap.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+:::tip NOTE
+
+**LDAP user provider 是 GreptimeDB 企业版的功能。**
+
+:::
+
+# LDAP user provider 配置
+
+GreptimeDB 可以使用外部 LDAP 服务以验证用户。我们从 [PostgreSQL](https://www.postgresql.org/docs/current/auth-ldap.html)
+中借鉴了类似的想法。在 GreptimeDB 中，LDAP 鉴权也分为两种模式："simple bind" 和 "search bind"。
```
Comment (sunng87) [inline i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/authentication/ldap.md:12]:
```suggestion
GreptimeDB 可以使用外部 LDAP 服务以验证用户。与 [PostgreSQL 中的 LDAP 机制相似](https://www.postgresql.org/docs/current/auth-ldap.html)， 在 GreptimeDB 中，LDAP 鉴权也分为两种模式："simple bind" 和 "search bind"。
```

File: i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/authentication/overview.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# 鉴权
+
+当用户尝试连接到数据库时，将会进行身份验证。在 GreptimeDB 中，用户通过“user provider”进行身份验证。GreptimeDB 中有多种 user
+provider 实现：
+
+- [Static user provider](./static.md)：一个简单的内置 user provider 实现，从静态文件中查找用户。
+- [LDAP user provider](./ldap.md)：使用外部 LDAP 服务进行用户身份验证。
```
Comment (sunng87) [inline i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/authentication/overview.md:7]:
```suggestion
- [LDAP user provider](./ldap.md)：**企业版功能。**使用外部 LDAP 服务进行用户身份验证。
```

File: i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/authentication/overview.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+:::tip NOTE
+
+**LDAP user provider 是 GreptimeDB 企业版的功能。**
+
+:::
```
Comment (sunng87) [inline i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/authentication/overview.md:13]:
```suggestion
```

**Comment by sunng87** [comment]
@nicecui we can plan a dedicated directory for Greptime Enterprise.

**Comment by MichaelScofield** [comment]
@sunng87 PTAL

File: i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/authentication/ldap.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
+# LDAP user provider
```
Comment (nicecui) [inline i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/authentication/ldap.md:1]:
```suggestion
# LDAP User Provider
```
Do we need to translate this title?

File: i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/authentication/ldap.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+# 进行 search bind 鉴权时，用于匹配的用户属性。
+# 如果未指定属性，则将使用 uid 属性。
+search_attribute = "cn"
+# 进行 search bind 鉴权时，使用的搜索过滤器。
+# "$username" 将被替换为客户端传来的用户名。
+# 这允许比 search_attribute 更灵活的用户搜索。
+search_filter = "(cn=$username)"
+```
+
+# 在 GreptimeDB 中使用 LDAP user provider
```
Comment (nicecui) [inline i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/authentication/ldap.md:66]:
```suggestion
## 在 GreptimeDB 中使用 LDAP User Provider
```

File: i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/authentication/ldap.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+要使用 LDAP user provider，首先参照上文配置你的 LDAP 鉴权模式，然后在启动 GreptimeDB 时使用 `--user-provider` 参数，将其设置为
```
Comment (nicecui) [inline i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/authentication/ldap.md:68]:
```suggestion
要使用 LDAP User Provider，首先参照上文配置你的 LDAP 鉴权模式，然后在启动 GreptimeDB 时使用 `--user-provider` 参数，将其设置为
```

File: i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/authentication/ldap.md [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+standalone GreptimeDB：
+
+```shell
+greptime standalone start --user-provider=ldap_user_provider:/home/greptimedb/ldap.toml
+```
+
+现在你就可以使用你的 LDAP 用户账户创建一个连接到 GreptimeDB 了。
+
+> 注意：如果你使用 mysql cli 连接到配置了 LDAP user provider 的 GreptimeDB，你需要在 mysql cli 中指定
+`--enable-cleartext-plugin`。
```
Comment (nicecui) [inline i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/authentication/ldap.md:79]:
```suggestion
:::tip 注意
如果你使用 MySQL CLI 连接到配置了 LDAP User Provider 的 GreptimeDB，你需要在 MySQL CLI 中指定
`--enable-cleartext-plugin`。
```

File: i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/authentication/overview.md [hunk 11]
```diff
@@ -0,0 +1,8 @@
+# 鉴权
+
+当用户尝试连接到数据库时，将会进行身份验证。在 GreptimeDB 中，用户通过“user provider”进行身份验证。GreptimeDB 中有多种 user
```
Comment (nicecui) [inline i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/authentication/overview.md:3]:
```suggestion
当客户端尝试连接到数据库时，将会进行身份验证。GreptimeDB 通过 “user provider” 进行身份验证。GreptimeDB 中有多种 user
```

File: i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/authentication/static.md [hunk 12]
```diff
@@ -1,6 +1,6 @@
-# 鉴权
+# Static user provider
```
Comment (nicecui) [inline i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/authentication/static.md:1]:
```suggestion
# Static User Provider
```

File: docs/user-guide/deployments/authentication/static.md [hunk 13]
```diff
@@ -1,6 +1,6 @@
-# Authentication
+# Static user provider
```
Comment (nicecui) [inline docs/user-guide/deployments/authentication/static.md:1]:
```suggestion
# Static User Provider
```

File: i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/authentication/ldap.md (hunk 7 above)
Comment (MichaelScofield) [reply i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/authentication/ldap.md:1]: I'd rather not. It's uncommon to translate "provider".

**Comment by MichaelScofield** [comment]
@nicecui PTAL

**Comment by nicecui** [comment]
> @nicecui we can plan a dedicated directory for Greptime Enterprise.

Got it

File: i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/authentication/ldap.md [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+
+```shell
+greptime standalone start --user-provider=ldap_user_provider:/home/greptimedb/ldap.toml
+```
+
+现在你就可以使用你的 LDAP 用户账户创建一个连接到 GreptimeDB 了。
+
+:::tip 注意
+如果你使用 MySQL CLI 连接到配置了 LDAP User Provider 的 GreptimeDB，你需要在 MySQL CLI 中指定
+`--enable-cleartext-plugin`。
```
Comment (nicecui) [inline i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/authentication/ldap.md:80]:
```suggestion
`--enable-cleartext-plugin`。
:::
```

**Review by nicecui — APPROVED** [review]

**Review by sunng87 — APPROVED** [review]

**Pull request merged**